test/cpp/end2end/client_lb_end2end_test.cc

```diff
@@ -798,6 +798,38 @@ TEST_F(ClientLbEnd2endTest, PickFirstUpdateSuperset) {
   EXPECT_EQ("pick_first", channel->GetLoadBalancingPolicyName());
 }
 
+TEST_F(ClientLbEnd2endTest, PickFirstUpdateToUnconnected) {
+  const int kNumServers = 4;
+  CreateServers(kNumServers);
+  auto response_generator = BuildResolverResponseGenerator();
+  auto channel = BuildChannel("pick_first", response_generator);
+  auto stub = BuildStub(channel);
+
+  std::vector<int> ports;
+
+  // Perform try to send rpc against a list where one server is available.
+  ports.emplace_back(servers_[0]->port_);
+  ports.emplace_back(servers_[1]->port_);
+  response_generator.SetNextResolution(ports);
+  gpr_log(GPR_INFO, "****** SET [0, 1] *******");
+  CheckRpcSendFailure(stub);
```

Comment:
Is it actually necessary to have an initially failing server list?  It seems like the bug happens when there is a valid server, followed by having no valid servers, followed by having a valid server again.  It's not clear to me what is added by having this failure initially.  I think we should be able to start out by just starting the server and then calling `CheckRpcSendOk()`.

Reply:
Done.